src/Core/Style.js

```diff
@@ -43,59 +43,180 @@ function getImageFromSprite(sprites, key) {
+ *
+ * const layer = new itowns.ColorLayer('foo', {
+ *      source: source,
+ *      style: style,
+ * });
+ *
+ * view.addLayer(layer);
  */
 class Style {
     /**
-     * Constructs the object.
-     * @param  {Object}  [params={}] An object that can contain all properties of a Style.
+     * @param {Object} [params={}] An object that can contain all properties of
+     * a Style.
+     *
      * @constructor
      */
     constructor(params = {}) {
         this.isStyle = true;
+
         params.fill = params.fill || {};
         params.stroke = params.stroke || {};
         params.point = params.point || {};
         params.text = params.text || {};
 
         this.fill = {
             color: params.fill.color,
-            opacity: params.fill.opacity,
+            opacity: params.fill.opacity == undefined ? 1.0 : params.fill.opacity,
             pattern: params.fill.pattern,
         };
+
         this.stroke = {
             color: params.stroke.color,
-            opacity: params.stroke.opacity,
-            width: params.stroke.width,
+            opacity: params.stroke.opacity == undefined ? 1.0 : params.stroke.opacity,
+            width: params.stroke.width == undefined ? 1.0 : params.stroke.width,
             dasharray: params.stroke.dasharray || [],
         };
+
         this.point = {
             color: params.point.color,
-            line: params.point.line,
-            opacity: params.point.opacity,
-            radius: params.point.radius,
-            width: params.point.width,
+            line: params.point.line || 'red',
+            opacity: params.point.opacity == undefined ? 1.0 : params.point.opacity,
+            radius: params.point.radius == undefined ? 2.0 : params.point.radius,
+            width: params.point.width || 0.0,
         };
+
         this.text = {
             field: params.text.field,
-            font: params.text.font,
-            color: params.text.color,
+            color: params.text.color || 'black',
+            anchor: params.text.anchor || 'center',
+            offset: params.text.offset || [0, 0],
+            padding: params.text.padding || 2,
+            size: params.text.size || 16,
+            placement: params.text.placement || 'point',
+            rotation: params.text.rotation || 'auto',
+            wrap: params.text.wrap || 10,
+            spacing: params.text.spacing || 0,
+            transform: params.text.transform || 'none',
+            justify: params.text.justify || 'center',
+            opacity: params.text.opacity || 1.0,
+            font: params.text.font || ['Open Sans Regular', 'Arial Unicode MS Regular', 'sans-serif'],
+            halo: {
+                color: (params.text.halo && params.text.halo.color) || 'black',
+                width: (params.text.halo && params.text.halo.width) || 0,
+                blur: (params.text.halo && params.text.halo.blur) || 0,
+            },
         };
     }
+
+    clone() {
```

Comment:
could add `copy`  and use it in `clone` function?

Reply:
Done